modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/SocketAccess.java

```diff
@@ -52,6 +52,10 @@ public static void doPrivilegedVoidException(StorageRunnable action) {
         }
     }
 
+    public static <E extends Exception> void doPrivilegedVoidExceptionExplicit(Class<E> exception, StorageRunnable action) throws E {
+        doPrivilegedVoidException(action);
+    }
```

Comment:
I added this because the `IOException` gets lost in the `doPrivelegedVoidException` call, and it means that even though the lambda throws an IOException that will propagate, intelliJ greys out the `throws IOException` on the method signature. Not sure if this is the right way to go about it.

Reply:
To explain myself, I was thinking a change to the existing method like the follows
```diff
--- a/modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/SocketAccess.java
+++ b/modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/SocketAccess.java
@@ -40,7 +40,7 @@ public final class SocketAccess {
         }
     }

-    public static void doPrivilegedVoidException(StorageRunnable action) {
+    public static void doPrivilegedVoidException(StorageRunnable action) throws IOException {
         SpecialPermission.check();
         try {
             AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
@@ -48,6 +48,9 @@ public final class SocketAccess {
                 return null;
             });
         } catch (PrivilegedActionException e) {
+            if (e.getCause() instanceof IOException ioException) {
+                throw ioException;
+            }
             Throwables.rethrow(e.getCause());
         }
     }
```
It requires cascading changes. So a separate PR feels better.

Reply:
Yeah I had considered that but it's very impactful as it would mean all calls to `doPrivelegedVoidException` would throw an `IOException`, even if the `StorageRunnable` did not, which is just a different kind of janky. I don't think there's a nice easy answer for this without implementing loads of different types of callback or something ugly.

Anyhow I think I'm OK with just leaving it. All of this stuff goes away soon anyhow I think (with security manager being deprecated)